media/CircleEditor/view-sidebar.js

```diff
@@ -538,37 +580,234 @@ class NodeAttributeView {
+                    if (type.toLowerCase() === this._attribute.type) {
+                        option.setAttribute('selected', 'selected');
+                    }
+                    this._select.appendChild(option);
+                }
+                typeLine.appendChild(this._select);
+                this._element.appendChild(typeLine);
+            }
+
+            const description = this._attribute.description;
+            if (description) {
+                const descriptionLine = this._host.document.createElement('div');
+                descriptionLine.className = 'sidebar-view-item-value-line-border';
+                descriptionLine.innerHTML = description;
+                this._element.appendChild(descriptionLine);
+            }
+        }
+
+        if (this._attribute.type === 'tensor' && value) {
+            const state = value.state;
+            const valueLine = this._host.document.createElement('div');
+            valueLine.className = 'sidebar-view-item-value-line-border';
+            const contentLine = this._host.document.createElement('pre');
+            contentLine.innerHTML = state || value.toString();
+            valueLine.appendChild(contentLine);
+            this._element.appendChild(valueLine);
+        }
+    }
+
+    save() {
+        if (this._attribute._type === 'int32') {
+            if (this._line.value - 0 > 2147483648) {
+                // TODO Enable posting message
+                // vscode.postMessage({
+                //     command : 'alert',
+                //     text: 'Can\'t exceed 2,147,483,648'
+                // });
+                return;
+            }
+        }
+
+        // TODO It will be implemented when the attribute type of the CustomOperator can be
+        // received. if (this._isCustom === true) {
+        //     const input = this._host.document.getElementById('attribute' + this._index);
+        //     if (this._select.value === 'int') {
+        //         if (this._line.value - 0 > 2147483648) {
+        //             // TODO Enable posting message
+        //             // vscode.postMessage({
+        //             //     command : 'alert',
+        //             //     text: 'Can\'t exceed 2,147,483,648'
+        //             // });
+        //             return;
+        //         }
+        //     }
+        //     input.disabled = true;
+        //     this._attribute._name = input.value;
+        //     this._attribute._type = this._select.value;
+        //     this._attribute._value = this._line.value;
+        // }
+
+        while (this._element.childElementCount) {
+            this._element.removeChild(this._element.lastChild);
+        }
+
+        if (!this._isCustom) {
+            this._editObject._attribute.name = this._attribute.name;
+            this._editObject._attribute._value = this._line.value;
+            this._editObject._attribute._type = this._attribute.type;
+        }
+        // } else {
```

Comment:
Why this is disabled?
This seems not related with message handling...

Reply:
key.type is undefined because the function that receives the attribute type of the custom operator is not implemented in the DS side yet.